config/v1/types_scheduling.go

```diff
@@ -48,6 +51,13 @@ type SchedulerSpec struct {
 	// would not be applied.
 	// +optional
 	DefaultNodeSelector string `json:"defaultNodeSelector,omitempty"`
+	// percentageOfMastersSchedulable allows a certain percentage of available master nodes to
+	// be schedulable. As of now, the supported values are 0 and 100. In future, we 
+	// will allow a certain percentage of master nodes to be schedulable. For example,
+	// if we have 3 running master nodes in the cluster, setting this value to 33% means, 1
+	// node of 3 available master nodes would become schedulable. The absolute value is calculated
+	// by rounding down. This field is optional
+	PercentageOfMastersSchedulable intstr.IntOrString `json:"percentageOfMastersSchedulable"`
```

Comment:
I think you need
`+optional`
at the end of the doc text (just like the above `DefaultNodeSelector`).
And also make it `*intstr.IntOrString` so we can distinguish unset or empty.

Reply:
The naming seems awkward; how about just `MaxMastersSchedulable` or so, a bit like `RollingUpdateDeployment`'s `MaxUnavailable`?

Reply:
> I think you need
+optional
at the end of the doc text (just like the above DefaultNodeSelector).

I think it's not needed as long as we provide some text. I have seen both the variants but not sure which one is preferred

> The naming seems awkward; how about just MaxMastersSchedulable or so, a bit like RollingUpdateDeployment's MaxUnavailable?

Yeah, I agree. I got to know about it while I was implementing, where it's possible to have both integer values and strings. Will change that in a bit along with comments.

Reply:
intOrString for explicitly specified percentages is strange. Why not just making it a float?

And add minimum and maximum as OpenAPI specification.

Reply:
We can add number of nodes as well, for example a user can explicitly tell `3` instead of 40%. This is the pattern we've been following in other apps.